src/components/TradeAssetSearch/hooks/useGetCustomTokenPriceQuery.tsx

```diff
@@ -0,0 +1,80 @@
+import { type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { useQueries, type UseQueryResult } from '@tanstack/react-query'
+import axios from 'axios'
+import { mergeQueryOutputs } from 'react-queries/helpers'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+
+type UseGetCustomTokenPricesQueryProps = {
+  assetIds: AssetId[]
+}
+
+type UseGetCustomTokenPricesQueryReturn = {
+  assetId: AssetId
+  marketData: ZerionMarketData | undefined
+}[]
+
+const ZERION_BASE_URL = 'https://zerion.shapeshift.com/'
+
+const axiosInstance = axios.create()
+
+type ZerionMarketData = {
+  price: number
+  total_supply: number
+  circulating_supply: number
+  fully_diluted_valuation: number
+  market_cap: number
+  changes: {
+    percent_1d: number
+    percent_30d: number
+    percent_90d: number
+    pecent_365d: number
+  }
+}
+
```

Comment:
could type what we care about from fungibles:
```ts
type ZerionFungibles = {
  data: {
    attributes: {
      market_data: ZerionMarketData
    }
  }
}
```

Reply:
https://github.com/shapeshift/web/pull/7370/commits/1c8f6da041d9491392285d0b470517a76cce306d